services/src/main/java/org/fao/geonet/api/records/MetadataInsertDeleteApi.java

```diff
@@ -215,8 +215,18 @@ public class MetadataInsertDeleteApi {
     @ResponseStatus(HttpStatus.NO_CONTENT)
     public void deleteRecord(
         @Parameter(description = API_PARAM_RECORD_UUID, required = true) @PathVariable String metadataUuid,
-        @Parameter(description = API_PARAM_BACKUP_FIRST, required = false) @RequestParam(required = false, defaultValue = "true") boolean withBackup,
+        @Parameter(description = API_PARAM_BACKUP_FIRST, required = false) @RequestParam(required = false) Boolean withBackup,
         HttpServletRequest request) throws Exception {
+        boolean isMdDeleteBackupEnable = settingManager.getValueAsBool(Settings.METADATA_DELETE_ENABLEBACKUP);
+        if (withBackup == null) {
+            Log.info(LOGGER, String.format("withBackup paramter is missing. Will use setting of metadata/delete/enablebackup '%s' during the deletion process", isMdDeleteBackupEnable));
+            withBackup = isMdDeleteBackupEnable;
+        } else if (withBackup != isMdDeleteBackupEnable) {
```

Comment:
This is not correct.
The current default is true which means that backups are enabled.  Before this fix, someone could call the api and specify that they don't want a backup and that would have worked.  i.e. maybe it was a test metadata record and when they delete it, they don't want a backup.  Now with this fix, it will fail if the default it true and someone tries to indicate that they don't want a backup.

Reply:
I made the logic to have three flag.

![image](https://github.com/user-attachments/assets/67a9b441-720f-4661-b319-50c60503a585)


i found it will include all cases we are discussing here. 

1) If the API parameter is not supplied (null), it will only flip it when we set ForceNoBack option in setting. Otherwise, remain as true as what it was before.

2) If the API parameter is supplied and set to true, detect if ForceNoBackup and throw an error for conflicting setup and client side call. Otherwise, do what the API parameter instructed.

3) If the API parameter is supplied and set to false, detect if ForceBackup and throw an error for conflicting setup and client side call. Otherwise, do what the API parameter instructed.